clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/IngestionBatchProcessor.java

```diff
@@ -123,38 +133,35 @@ public List<PubSubMessageProcessedResultWrapper<KafkaKey, KafkaMessageEnvelope,
     if (records.isEmpty()) {
       return Collections.emptyList();
     }
-    AtomicBoolean isAllMessagesFromRTTopic = new AtomicBoolean(true);
+    boolean isAllMessagesFromRTTopic = true;
     List<PubSubMessageProcessedResultWrapper<KafkaKey, KafkaMessageEnvelope, Long>> resultList =
         new ArrayList<>(records.size());
-    records.forEach(r -> {
-      resultList.add(new PubSubMessageProcessedResultWrapper<>(r));
-      if (!r.getTopicPartition().getPubSubTopic().isRealTime()) {
-        isAllMessagesFromRTTopic.set(false);
-      }
-    });
-    if (!isWriteComputationEnabled && !isActiveActiveReplicationEnabled) {
-      return resultList;
-    }
-    // Only handle records from the real-time topic
-    if (!isAllMessagesFromRTTopic.get()) {
-      return resultList;
-    }
-
     /**
      * We would like to process the messages belonging to the same key sequentially to avoid race conditions.
      */
     int totalNumOfRecords = 0;
     Map<ByteArrayKey, List<PubSubMessageProcessedResultWrapper<KafkaKey, KafkaMessageEnvelope, Long>>> keyGroupMap =
         new HashMap<>(records.size());
 
-    for (PubSubMessageProcessedResultWrapper<KafkaKey, KafkaMessageEnvelope, Long> r: resultList) {
-      PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> message = r.getMessage();
-      if (!message.getKey().isControlMessage()) {
+    for (PubSubMessage<KafkaKey, KafkaMessageEnvelope, Long> message: records) {
+      if (!message.getTopicPartition().getPubSubTopic().isRealTime()) {
```

Comment:
should we check for not control message here?

Reply:
Emm.. I don't think there is a need and we don't have a use case where the CM is from VT, but actual messages are from RT, right?